src/ProtectedProxies/MinimalProtectedProxyFactory.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: UNLICENSED
+// (c) SphereX 2023 Terms&Conditions
+
+pragma solidity ^0.8.0;
+
+import {SphereXProtected} from "../SphereXProtected.sol";
+import {SpherexProtetedMinimalProxy} from "./ProtectedMinimalProxy.sol";
+
+// TODO: add event for deploying
+
+/**
+ * @title A factory contract that deploys the SpherexProtetedMinimalProxy contracts.
+ */
+contract MinimalProtectedProxyFactory is SphereXProtected {
+
+    /**
+     * Deploys a new protected minimal proxy. Sets the inital protected values (admin, operator and engie) to equal those of the factory.
+     *  If the current engine is not the null address, also updates the engine about a valid new sender
+     * @dev notice the deploy function itself is NOT PROTECTED
+     */
+    function deploy(address implementation) virtual public returns (address) {
```

Comment:
regarding the protectedsigs, i dont think it is the factory responsability to call it, since it doesnt know the abi of the implementation, so the one calling the deploy function should call it, but the operator of the proxy is the same operator of the factory, and it doesnt necessarily the one calling the factory, so we can make the deploy function receive the admin\engine\operator as arguments and this way the one calling it will be in charge and in zeebu we can make their factory be the operator and after it sets the protectedsigs give the role to the operator of the system